### PR TITLE
Confirmation before uninstalling a plugin

### PR DESCRIPTION
**Pull request by FragDev**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Added a confirmation modal before uninstalling the plugin.
Both "plugins" and "marketplace" views have been taken into account.

image

### CODE REVIEW

File: templates/components/plugin_uninstall_modal.html.twig [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+<!-- Modal -->
+<div class="modal fade" id="uninstallModal" tabindex="-1" role="dialog" aria-hidden="true">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <a type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></a>
+            <div class="modal-status bg-danger"></div>
+            <div
+                class="modal-body text-center py-4">
+                <i class="fa-2x ti ti-alert-triangle mb-2 text-danger"></i>
+                <h3>{{ _x('uninstall_plugin', 'Are you sure?') }}</h3>
```
Comment (cedric-anne) [inline templates/components/plugin_uninstall_modal.html.twig:46]: No need to have context here (we rarely use contexts).
```suggestion
                <h3>{{ __('Are you sure?') }}</h3>
```

File: templates/components/plugin_uninstall_modal.html.twig [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                <div class="text-muted">{{ _x('uninstall_plugin', 'By uninstalling the plugin you will lose all the data of the plugin.') }}</div>
```
Comment (cedric-anne) [inline templates/components/plugin_uninstall_modal.html.twig:47]:
```suggestion
                <div class="text-muted">{{ __('By uninstalling the plugin you will lose all the data of the plugin.') }}</div>
```

File: templates/components/plugin_uninstall_modal.html.twig [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # ---------------------------------------------------------------------
+ #}
+
+<!-- Button trigger modal -->
+{{ open_btn|raw }}
```
Comment (cedric-anne) [inline templates/components/plugin_uninstall_modal.html.twig:35]: I do not like the idea of defining the whole button HTML on PHP side, but as both buttons are really different on both marketplace and legacy plugin list, it is OK for me.

I guess we will find time to refactor these views in next major versions, in order to keep only one of them.

File: templates/components/plugin_uninstall_modal.html.twig (hunk 2 above)
Comment (cedric-anne) [inline templates/components/plugin_uninstall_modal.html.twig:47]: Also, the plugin name should be added in this sentence, so the user will be sure that he is not deleting the wrong plugin data.

**Review by orthagh — CHANGES_REQUESTED** [review]
There's an issue in the former plugin list. 
The plugin name mentioned in the modal take the first installed entry of the list and not the one selected for uninstallation.

Ex, I clicked on uninstall for Gantt plugin:
image

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**